testsuite/oauth/smart.py

```diff
@@ -0,0 +1,105 @@
+""" The SMART oAuth strategy. """
+import requests
+
+
+ERROR_AUTH_REQUEST = """
+Authenticated POST to {url} returned {status_code}.
+{text}
+"""
+
+
+class SmartStrategy(object):
+    """ Implements the lib.oauth.Strategy interface. """
```

Comment:
I'd like it to work for everyone, but right now I've only tested it against our reference server. What will probably happen is that I'll break out a set of "authorize" strategies as well, and each will be responsible for managing either a headless authorize path (like ours) or some kind of webdriver/session based path.